bundles/org.openhab.binding.carnet/src/main/java/org/openhab/binding/carnet/internal/handler/CarNetVehicleHandler.java

```diff
@@ -0,0 +1,643 @@
+    private final Map<String, Object> channelData = new HashMap<>();
+    private final CarNetTokenManager tokenManager;
+
+    public String thingId = "";
+    private CarNetApi api = new CarNetApi();
+    private @Nullable CarNetAccountHandler accountHandler;
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private int updateCounter = 0;
+    private int skipCount = 1;
+    private boolean forceUpdate;
+    private boolean channelsCreated = false;
+    private boolean testData = false;
+
+    private Map<String, CarNetVehicleBaseService> services = new LinkedHashMap<>();
+    private CarNetServiceAvailability serviceAvailability = new CarNetServiceAvailability();
+    private CarNetCombinedConfig config = new CarNetCombinedConfig();
+
+    public CarNetVehicleHandler(Thing thing, CarNetTextResources resources, CarNetIChanneldMapper idMapper,
+            CarNetTokenManager tokenManager) {
+        super(thing);
+
+        this.resources = resources;
+        this.idMapper = idMapper;
+        this.tokenManager = tokenManager;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing!");
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Register listener and wait for account being ONLINE
+        CarNetAccountHandler handler = null;
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            handler = (CarNetAccountHandler) bridge.getHandler();
+        }
+        if ((handler == null)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED,
+                    "Account Thing is not initialized!");
+            return;
+        }
+        accountHandler = handler;
+        thingId = getThing().getUID().getId();
+
+        api = new CarNetApi(handler.getHttpClient(), tokenManager);
+
+        scheduler.schedule(() -> {
+            if (accountHandler != null) {
+                accountHandler.registerListener(this);
+                setupPollingJob();
+            }
+        }, 1, TimeUnit.SECONDS);
+    }
+
+    /**
+     * (re-)initialize the thing
+     *
+     * @return true=successful
+     */
+    boolean initializeThing() {
+        channelData.clear(); // clear any cached channels
+        boolean successful = true;
+        String error = "";
+        try {
+            if (accountHandler != null) {
+                config = accountHandler.getCombinedConfig();
+            }
+            config.vehicle = getConfigAs(CarNetVehicleConfiguration.class);
+            Map<String, String> properties = getThing().getProperties();
+            skipCount = Math.max(config.vehicle.refreshInterval / POLL_INTERVAL_SEC, 2);
+            channelsCreated = false;
+
+            String vin = "";
+            if (properties.containsKey(PROPERTY_VIN)) {
+                vin = properties.get(PROPERTY_VIN);
+            }
+            if ((vin == null) || vin.isEmpty()) {
+                logger.info("VIN not set (Thing properties)");
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.